modules/tinymce/src/core/main/ts/fmt/CaretFormat.ts

```diff
@@ -165,6 +165,29 @@ const cleanFormatNode = (editor: Editor, caretContainer: Node, formatNode: Eleme
   }
 };
 
+const normalizeNbspsBetween = (editor: Editor, caretContainer: Node | null) => {
+  const handler = () => {
+    if (caretContainer && !editor.dom.isEmpty(caretContainer)) {
+      Traverse.prevSibling(SugarElement.fromDom(caretContainer)).each((node) => {
+        if (NodeType.isText(node.dom)) {
+          node.dom.data = node.dom.data.replace(Unicode.nbsp, ' ');
```

Comment:
You can use sugars Text.get/set here I think.

Reply:
changed using: `SugarText.get/set`